FWCore/Concurrency/interface/WaitingTaskHolder.h

```diff
@@ -48,6 +48,12 @@ namespace edm {
       return *this;
     }
 
+    WaitingTaskHolder& operator=(WaitingTaskHolder&& iRHS) {
+      WaitingTaskHolder tmp(std::move(iRHS));
```

Comment:
std::move is unnecessary, as iRHS is already a && ?

Reply:
I don't believe that is guaranteed

Reply:
I did a quick test with compiler explorer using -O0 and the compiler used the copy constructor if I did not do an explicit `std::move`

Reply:
My understanding (could be wrong) is that the `&&` in `iRHS` is about to what `iRHS` can bind to, but the `iRHS` itself is an lvalue (or something similar, but not rvalue).

Reply:
I tested using -O2 and -O3 and also got the same result.